packages/components/src/confirm-dialog/stories/index.js

```diff
@@ -15,128 +10,129 @@ import Button from '../../button';
-		</>
-	);
-};
 
-export const WithJSXMessage = () => {
-	const [ confirmVal, setConfirmVal ] = useState( "Hasn't confirmed yet" );
-
-	return (
-		<>
-			<ConfirmDialog onConfirm={ () => setConfirmVal( 'Confirmed!' ) }>
-				<Heading level={ 2 }>{ daText() }</Heading>
-			</ConfirmDialog>
 			<Heading level={ 1 }>{ confirmVal }</Heading>
 		</>
 	);
 };
 
-export const VeeeryLongMessage = () => {
-	const [ confirmVal, setConfirmVal ] = useState( "Hasn't confirmed yet" );
+// Simplest usage: just declare the component with the required `onConfirm` prop. Note: the `onCancel` prop is optional here, unless you'd like to render the component in Controlled mode (see below)
+export const _default = Template.bind( {} );
+_default.args = {};
 
-	return (
-		<>
-			<ConfirmDialog onConfirm={ () => setConfirmVal( 'Confirmed!' ) }>
-				{ daText().repeat( 20 ) }
-			</ConfirmDialog>
-			<Heading level={ 1 }>{ confirmVal }</Heading>
-		</>
-	);
-};
-
-export const UncontrolledAndWithExplicitOnCancel = () => {
-	const [ confirmVal, setConfirmVal ] = useState(
-		"Hasn't confirmed or cancelled yet"
-	);
-
-	return (
-		<>
-			<ConfirmDialog
-				onConfirm={ () => setConfirmVal( 'Confirmed!' ) }
-				onCancel={ () => setConfirmVal( 'Cancelled' ) }
-			>
-				{ daText() }
-			</ConfirmDialog>
-			<Heading level={ 1 }>{ confirmVal }</Heading>
-		</>
-	);
+// To customize button text, pass the `cancelButtonText` and/or `confirmButtonText` props.
+export const withCustomButtonLabels = Template.bind( {} );
+withCustomButtonLabels.args = {
+	cancelButtonText: 'No thanks',
+	confirmButtonText: 'Yes please!',
 };
 
-// Controlled `ConfirmDialog`s require both `onConfirm` *and* `onCancel to be passed
-// It's expected that the user will then use it to hide the dialog, too (see the
-// `setIsOpen` calls below).
-export const Controlled = () => {
+const controlledSnippet = `() => {
 	const [ isOpen, setIsOpen ] = useState( false );
-	const [ confirmVal, setConfirmVal ] = useState(
-		"Hasn't confirmed or cancelled yet"
-	);
-
+	const [ confirmVal, setConfirmVal ] = useState('');
+	
 	const handleConfirm = () => {
 		setConfirmVal( 'Confirmed!' );
 		setIsOpen( false );
 	};
-
+	
 	const handleCancel = () => {
 		setConfirmVal( 'Cancelled' );
 		setIsOpen( false );
 	};
-
+	
 	return (
 		<>
-			<ConfirmDialog
-				isOpen={ isOpen }
-				onConfirm={ handleConfirm }
-				onCancel={ handleCancel }
-			>
-				{ daText() }
-			</ConfirmDialog>
-
-			<Heading level={ 1 }>{ confirmVal }</Heading>
-
-			<Button variant="primary" onClick={ () => setIsOpen( true ) }>
-				Open ConfirmDialog
-			</Button>
+		<ConfirmDialog
+		isOpen={ isOpen }
+		onConfirm={ handleConfirm }
+		onCancel={ handleCancel }
+		>
+		Would you like to privately publish the post now?
+		</ConfirmDialog>
+		
+		<Heading level={ 1 }>{ confirmVal }</Heading>
+		
+		<Button variant="primary" onClick={ () => setIsOpen( true ) }>
+		Open ConfirmDialog
+		</Button>
```

Comment:
The indentation on this code snippet looks a bit off (although feel free to ignore in case we decide to remove the `Controlled` story altogether)

```suggestion
			<ConfirmDialog
				isOpen={ isOpen }
				onConfirm={ handleConfirm }
				onCancel={ handleCancel }
			>
				Would you like to privately publish the post now?
			</ConfirmDialog>
	
			<Heading level={ 1 }>{ confirmVal }</Heading>
		
			<Button variant="primary" onClick={ () => setIsOpen( true ) }>
				Open ConfirmDialog
			</Button>
```

Reply:
good catch, I didn't spot that in the template literal. I've fixed it, and also replaced the tabs with double spaces... it looks like Storybook is using double space indentation when it autogenerates from the template, so my tabs looked off when a manual snippet was rendered just above an auto generated one. This way they all match up in terms of formatting.